godot-core/src/builtin/projection.rs

```diff
@@ -0,0 +1,443 @@
+    pub fn inverse(self) -> Self {
+        self.glam(|mat| mat.inverse())
+    }
+
+    /// Returns `true` if this Projection performs an orthogonal projection.
+    ///
+    /// _Godot equivalent: Projection.is_orthogonal()_
+    pub fn is_orthogonal(&self) -> bool {
+        self.as_inner().is_orthogonal()
+    }
+
+    /// Returns a Projection with the X and Y values from the given [`Vector2`]
+    /// added to the first and second values of the final column respectively.
+    ///
+    /// _Godot equivalent: Projection.jitter_offseted()_
+    pub fn jitter_offseted(&self, offset: Vector2) -> Self {
+        self.as_inner().jitter_offseted(offset)
+    }
+
+    /// Returns a Projection with the near clipping distance adjusted to be
+    /// `new_znear`.
+    ///
+    /// Note: The original Projection must be a perspective projection.
+    ///
+    /// _Godot equivalent: Projection.perspective_znear_adjusted()_
+    pub fn perspective_znear_adjusted(&self, new_znear: f64) -> Self {
+        self.as_inner().perspective_znear_adjusted(new_znear)
+    }
+
+    #[doc(hidden)]
+    pub(crate) fn as_inner(&self) -> InnerProjection {
+        InnerProjection::from_outer(self)
+    }
+}
+
+impl From<Transform3D> for Projection {
+    fn from(trans: Transform3D) -> Self {
+        trans.glam(Mat4::from)
+    }
+}
+
+impl Default for Projection {
+    fn default() -> Self {
+        Self::IDENTITY
+    }
+}
+
+impl Mul for Projection {
+    type Output = Self;
+
+    fn mul(self, rhs: Self) -> Self::Output {
+        self.glam2(&rhs, |a, b| a * b)
+    }
+}
+
+impl Mul<Vector4> for Projection {
+    type Output = Vector4;
+
+    fn mul(self, rhs: Vector4) -> Self::Output {
+        self.glam2(&rhs, |m, v| m * v)
+    }
+}
+
+impl GlamType for Mat4 {
+    type Mapped = Projection;
+
+    fn to_front(&self) -> Self::Mapped {
+        Projection {
+            cols: [
+                self.col(0).to_front(),
+                self.col(1).to_front(),
+                self.col(2).to_front(),
+                self.col(3).to_front(),
+            ],
+        }
+    }
+
+    fn from_front(mapped: &Self::Mapped) -> Self {
+        Self {
+            x_axis: mapped.cols[0].to_glam(),
+            y_axis: mapped.cols[1].to_glam(),
+            z_axis: mapped.cols[2].to_glam(),
+            w_axis: mapped.cols[3].to_glam(),
+        }
+    }
+}
+
+impl GlamConv for Projection {
+    type Glam = Mat4;
+}
+
+impl GodotFfi for Projection {
+    ffi_methods! { type sys::GDExtensionTypePtr = *mut Self; .. }
+}
+
+/// A projections clipping plane.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub enum ProjectionPlane {
+    Near = 0,
+    Far = 1,
+    Left = 2,
+    Top = 3,
+    Right = 4,
+    Bottom = 5,
+}
+
+/// The eye to create a projection for, when creating a projection adjusted
+/// for head-mounted displays.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub enum Eye {
+    Left = 1,
+    Right = 2,
+}
```

Comment:
Great abstraction!

Keep in mind that we re-export all those identifiers via prelude, so it might be worth adding context. This could be named `ProjectionEye` for example.